SwiftyStoreKit/InAppReceipt.swift

```diff
@@ -30,10 +30,28 @@ public typealias ReceiptInfo = [String: AnyObject]
 
 // MARK: - Enumeration
 extension SwiftyStoreKit {
+    public enum PurchaseType {
+        // A consumable In-App Purchase must be purchased every time the user downloads it.
+        case Consumable
+        // Non-consumable In-App Purchases only need to be purchased once by users.
+        case NonConsumable
+        // Automatically renewable subscriptions allow the user to purchase updating and dynamic content for a set duration of time.
+        case AutomaticallyRenewableSubscription(validUntilDate: NSDate)
+        // Free subscriptions don’t expire and can only be offered in apps that are in the Magazines & Newspapers category.
+        case FreeSubscription
+        // Non-Renewing Subscriptions allow the sale of services with a limited duration.
+        case NonRenewingSubscription(validUntilDate: NSDate)
+    }
     public enum VerifyReceiptResult {
         case Success(receipt: ReceiptInfo)
         case Error(error: ReceiptError)
     }
+  
+    public enum VerifyPurchaseResult {
+        case Purchased(expiresDate: NSDate?)
+        case Expired(expiresDate: NSDate)
+        case NotPurchased
+    }
```

Comment:
Should we introduce separate result enums depending on whether we are dealing with Consumable/NonConsumable or Subscriptions?